Library/Homebrew/formula_installer.rb

```diff
@@ -863,12 +864,17 @@ def link(keg)
   end
 
   def install_plist
-    return unless formula.plist
+    return unless plist = formula.plist
 
-    formula.plist_path.atomic_write(formula.plist)
-    formula.plist_path.chmod 0644
-    log = formula.var/"log"
-    log.mkpath if formula.plist.include? log.to_s
+    plist = { formula.plist_name => plist } if plist.is_a? String
+
+    plist.each do |label, xml|
+      plist_path = formula.plist_path.sub(formula.plist_path.stem, label)
```

Comment:
Can you explain this a bit? Not totally clear what it's doing and why. A code comment might be good if it stays the same.

Reply:
Because the `plist_path`(`.stem`) needs to match the `label`, and the unique `label` is passed in here… so it needs to be substituted here. Directly reassigning `formula.plist_path =` here throws an error, and I think the only way to modify `formula.plist_path` before this point is to redefine `def plist_path` inside the `def plist` instance method (something which was frowned on before) Same goes for `plist_name` also…

If any of this begs the question "why?"… it's likely because I had to impliment some obscure workaround just to get it to work within the limiting constraints of this OOP architecture. It feels more like trying to play whack-a-mole wearing a straightjacket in the dark than writing a simple feature! 😅

Reply:
> Because the `plist_path`(`.stem`) needs to match the `label`, and the unique `label` is passed in here

So the `plist_path` is basically returning the wrong information because it needs to match the label?